File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.py
import sys

N = int(sys.stdin.readline().strip())
names = set()
for i in range(N):
  name, check = sys.stdin.readline().split()
  if name in names:
    names.remove(name)
  else:
    names.add(name)
    
name_list = list(names)
name_list.sort(reverse=True)
for i in name_list:
  print(i)

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        String s = bf.readLine();

        int N = Integer.parseInt(s);
        Set<String> set = new TreeSet<>();

        for(int i=0; i<N; i++){
            String[] sa = bf.readLine().split(" ");
            if("enter".equals(sa[1])) set.add(sa[0]);
            else set.remove(sa[0]);
        }

        List<String> list = new ArrayList<>(set);

        Collections.sort(list, new Comparator<String>() {
            @Override
            public int compare(String o1, String o2) {
                return o2.compareTo(o1);
            }
        });

        StringBuilder sb = new StringBuilder();
        for(String name : list){
            sb.append(name).append("\n");
        }

        System.out.println(sb);

    }
}
